lib/src/spec/common.dart

```diff
@@ -74,31 +75,189 @@ abstract class StatsMessageTraffic {
+  ///
+  /// Timestamps, in conjunction with the nonce,
+  /// are used to prevent requests from being replayed
+  ///
+  /// ref: TK2d
   String nonce;
+
+  /// The timestamp (in millis since the epoch) of this request.
+  ///
+  ///	Timestamps, in conjunction with the nonce, are used to prevent
+  ///	token requests from being replayed.
+  ///
+  /// ref: TK2d
   DateTime timestamp;
+
+  /// Requested time to live for the token.
+  ///
+  /// If the token request is successful, the TTL of the returned
+  /// token will be less than or equal to this value depending on
+  /// application settings and the attributes of the issuing key.
+  ///
+  /// 0 means Ably will set it to the default value
+  ///
+  /// Ref: TK2a
   int ttl;
+
+  TokenParams({
+    this.capability,
+    this.clientId,
+    this.nonce,
+    this.timestamp,
+    this.ttl
+  });
+
 }
 
+/// TokenDetails is a type containing the token request
+///
+/// response from the REST requestToken endpoint
+///
+/// spec: https://docs.ably.io/client-lib-development-guide/features/#TD1
 class TokenDetails {
-  TokenDetails(this.token);
-  String capability;
-  String clientId;
+
+  /// TD2
+  String token;
+
+  /// Token expiry time in milliseconds
+  ///
+  /// TD3
   int expires;
+
+  /// the time the token was issued in milliseconds
+  ///
+  /// TD4
   int issued;
-  String token;
-}
 
-abstract class TokenRequest {
+  /// stringified capabilities JSON
+  ///
+  /// TD5
   String capability;
+
+  /// clientId assigned to the token.
+  ///
+  /// If clientId is not set (i.e. null), then the token is prohibited
+  /// from assuming a clientId in any operations, however if clientId
+  /// is a wildcard string '*', then the token is permitted to assume
+  /// any clientId. Any other string value for clientId implies that the
+  /// clientId is both enforced and assumed for all operations for this token
+  ///
+  /// TD6
   String clientId;
+
+  TokenDetails(this.token, {
+    this.expires,
+    this.issued,
+    this.capability,
+    this.clientId
+  });
+
+  /// TD7
+  TokenDetails.fromMap(Map<String, dynamic> map){
+    token = map['token'] as String;
+    expires = map['expires'] as int;
+    issued = map['issued'] as int;
+    capability = map['capability'] as String;
+    clientId = map['clientId'] as String;
+  }
+
+}
+
+/// TokenRequest is a type containing the token request
+/// details sent to the REST requestToken endpoint
+///
+/// spec: https://docs.ably.io/client-lib-development-guide/features/#TE1
+class TokenRequest {
+
+  /// The keyName of the key against which this request is made.
+  ///
+  /// TE2
   String keyName;
-  String mac;
+
+  /// An opaque nonce string of at least 16 characters to ensure
+  ///	uniqueness of this request. Any subsequent request using the
+  ///	same nonce will be rejected.
+  ///
+  /// TE2, TE5
   String nonce;
+
+  /// The Message Authentication Code for this request. See the Ably
```

Comment:
There should be an empty line before `See the Ably ...`

Reply:
28208cf1cc513eb384ca2a99b314abfa74672508